data.base.api/src/main/java/ru/tinkoff/qa/neptune/data/base/api/queries/jdoql/JDOParameters.java

```diff
@@ -0,0 +1,144 @@
+package ru.tinkoff.qa.neptune.data.base.api.queries.jdoql;
+
+
+import ru.tinkoff.qa.neptune.data.base.api.PersistableObject;
+
+import javax.jdo.JDOQLTypedQuery;
+import javax.jdo.query.BooleanExpression;
+import javax.jdo.query.OrderExpression;
+import javax.jdo.query.PersistableExpression;
+import java.lang.reflect.Method;
+import java.util.function.Function;
+
+import static com.google.common.base.Preconditions.checkArgument;
+import static com.google.common.base.Preconditions.checkNotNull;
+import static java.lang.String.format;
+import static java.lang.reflect.Modifier.isStatic;
+import static java.util.Objects.nonNull;
+import static java.util.Optional.ofNullable;
+import static org.apache.commons.lang3.ArrayUtils.add;
+
+/**
+ * This class is designed to construct/detail {@link JDOQLTypedQuery}
+ *
+ * @param <T> is a type of {@link PersistableObject} to be selected
+ * @param <Q> is a type of {@link PersistableExpression} that represents {@code T} in a query
+ */
+@SuppressWarnings("unchecked")
+public abstract class JDOParameters<T extends PersistableObject, Q extends PersistableExpression<T>, S extends JDOParameters<T, Q, S>> {
+
+    final Q persistableExpression;
+    BooleanExpression where;
+    OrderExpression<?>[] orderExpressions;
+    Integer rangeStart;
+    Integer rangeEnd;
+
+    JDOParameters(Class<? extends Q> toUse) {
```

Comment:
Should it be private or protected?

Reply:
it think packege-private is convenient here